src/dune_rules/melange_stanzas.ml

```diff
@@ -47,6 +48,15 @@ module Emit = struct
     t
 
   let decode =
+    let extension_field name =
+      let+ field = field_o name (located string) in
+      match field with
+      | None -> ".js"
+      | Some (loc, extension) ->
+        if String.contains extension '.' then
+          User_error.raise ~loc [ Pp.textf "extension must not contain '.'" ];
```

Comment:
Why is that? An extension like `.bs.js` is invalid?

Reply:
In `bsconfig.json`, we expect users to write their extension starting with `.`. I think we should do the same here.

Reply:
That would be inconsistent with the extension field in dialects.

Reply:
I see. the alternative is to just check that it doesn't start with `.` then.

Reply:
Checked for dots on starting position + added tests in https://github.com/ocaml/dune/pull/6457/commits/d33f19cda8ad599056f993ed691da905f09125f7.